### ISSUE TITLE
Option to guess at better pathings?

### ISSUE THREAD

**Issue by NullVoxPopuli** [emberjs/ember-inspector#2425]
**Is your feature request related to a problem? Please describe.**
Was just poking around and noticed this:
![image](https://github.com/emberjs/ember-inspector/assets/199018/599545f6-7bc5-43c2-899e-e71abae827f5)
that path for the component is just hilarious -- almost goes across my whole screen  lololol

**Describe the solution you'd like**

Maybe an option in the inspector settings to provide the module path, rather than path on disk?

**Describe alternatives you've considered**
keep it silly

**Comment by patricklx** [comment]
what could be the best way here?
The nicest one would be to check for 'node_modules' and if its there, take the path part after it.
But that would also no be entirely correct, so maybe take the last part to show as template path and add a tooltip to show the full path?

though embroider should already take care of the leading path to app dir:
https://github.com/embroider-build/embroider/blob/main/packages/shared-internals/src/hbs-to-js.ts#L30.

or should this be fixed in embroider to ouput the same as classic builds?

**Comment by ef4** [comment]
The key question here is: why do we think templates need this special form of debugging info to point back at their source, when we don't expect the same from every javascript module? 

That distinction used to make sense, because templates used to be entirely separate from javascript. But now they are just values inside javascript. moduleName isn't even enough to uniquely locate a template anymore, given that you can have dozens of template tags in one module.

The inspector feature here should be more like a "jump to source" button. If instead of embedding `moduleName` next to each compiled template we instead embedded a tiny function:

```diff
t.createTemplateFactory)({
        id: "efyYydg8",
        block: '[[[8,[39,0],null,null,null]],[],false,["four-oh-four"]]',
-       moduleName: "pdpm-calc/templates/404.hbs",
+       debugHandle: function() {},
        isStrictMode: !1
    })
```

then at runtime the inspector can do `window.inspect(theComponent.debugHandle)` and it will immediately open the Sources pane of the dev tools pointing directly at the right component.

(I think it has to be a function because `window.inspect` is good at jumping to sources of functions, but it can't necessarily do the same for other kinds of objects.)

**Comment by RobbieTheWagner** [comment]
@ef4 I would be open to that.

**Comment by patricklx** [comment]
@ef4 good idea, but then moduleName should not be set if nothing is passed. Currently it sets it to a long path

**Comment by ef4** [comment]
I'm creating an RFC issue to centralize discussion of this problem: https://github.com/emberjs/rfcs/issues/940

**Comment by lifeart** [comment]
Wee may still need `moduleName` to support hot module replacement for components (in vite), at the moment we map changed component files to rendered components by `moduleName` resolved from `getComponentTemplate` or component itself
